week-8-practise/3-classes.js

```diff
@@ -29,14 +29,92 @@
  */
 
 class House {
-  constructor(windows, colour) {
-    this.windows = windows
+  constructor(windows, colour = "blue", numberOfRooms, location) {
```

Comment:
Although we haven't covered this yet, default parameters should be right-most, because JS maps the value to the argument by position.

Reply:
@sansaid please do you mind clarifying what you mean?

Reply:
Sure! So you used `colour="blue"` to set colour to have a default value. The only problem is that default values need to be one of the last arguments. So instead of `constructor(windows, colour = "blue", numberOfRooms, location)`, you should have `constructor(windows, numberOfRooms, locatoin, colour = "blue")` - non-default arguments on the left, default arguments on the right. 

The reason for this is that when you pass values into the function but don't want to pass a value for the `colour`, Javascript will always map the second argument to `colour`. However, if you positioned the default parameters last, then all you have to do is leave the last arguments out and Javascript will know to use the default values.

Reply:
Noted!  Thanks for the clarification, will update that line of code.